### ISSUE TITLE
when i use "$a=1$$b=2$"，the result is "$a=1b=2$"，but it is correct in mathjax, please see screenshot

### ISSUE THREAD

**Issue by xiewl** [KaTeX/KaTeX#1984]
this is options with katex
`window.renderMathInElement(dom, {
            delimiters: [
                { left: '$$', right: '$$', display: true },
                { left: '$', right: '$', display: false },
                { left: '\\[', right: '\\]', display: true },
                { left: '\\(', right: '\\)', display: false },
            ]
        });`
this is mathjax result
<img width="476" alt="WX20190524-182542@2x" src="https://user-images.githubusercontent.com/15881722/58321664-5b41cb00-7e51-11e9-98b9-9b4cf6f43355.png">

this is result with my katex code
<img width="744" alt="WX20190524-182722@2x" src="https://user-images.githubusercontent.com/15881722/58321770-9fcd6680-7e51-11e9-8fd1-544bf8ba7201.png">

**Comment by ronkok** [comment]
@xiewl You can get the desired behavior with:
```
  renderMathInElement(document.body, {delimiters:[
      {left: "$", right: "$", display: false},
      {left: "$$", right: "$$", display: true},
      {left: "\\(", right: "\\)", display: false},
      {left: "\\[", right: "\\]", display: true}
  ]});
```

**Comment by xiewl** [comment]
> @xiewl You can get the desired behavior with:
> 
> ```
>   renderMathInElement(document.body, {delimiters:[
>       {left: "$", right: "$", display: false},
>       {left: "$$", right: "$$", display: true},
>       {left: "\\(", right: "\\)", display: false},
>       {left: "\\[", right: "\\]", display: true}
>   ]});
> ```

i try it before, it does not work

**Comment by edemaine** [comment]
I believe the issue here is the adjacent $'s, which is related to #1830. I do believe this is correct LaTeX, so makes sense to fix (assuming it gets confirmed).

**Comment by ronkok** [comment]
The code I recommended gives me a good result. @xiewl Can you direct us to the URL of a working page? It might help with the diagnosis.

**Comment by xiewl** [comment]
> The code I recommended gives me a good result. @xiewl Can you direct us to the URL of a working page? It might help with the diagnosis.

this is a simple demo:  http://jsfiddle.net/d1uzt2wg/     
when "$”and"$” is together，it is not normal，but the same html is normal with mathjax

**Comment by ronkok** [comment]
The fix is to change the order in which the delimiters are specified. In your fiddle, use
```
     { left: '$', right: '$', display: false },
    { left: '$$', right: '$$', display: true },
```
instead of 
```
    { left: '$$', right: '$$', display: true },
     { left: '$', right: '$', display: false },
```

**Comment by xiewl** [comment]
> The fix is to change the order in which the delimiters are specified. In your fiddle, use
> 
> ```
>      { left: '$', right: '$', display: false },
>     { left: '$$', right: '$$', display: true },
> ```
> 
> instead of
> 
> ```
>     { left: '$$', right: '$$', display: true },
>      { left: '$', right: '$', display: false },
> ```

look at this
![QQ截图20190524235850](https://user-images.githubusercontent.com/15881722/58341066-f6ea3000-7e7f-11e9-9bad-49b7c0c72e47.png)

**Comment by ronkok** [comment]
Ah yes. I retract my comment. You have found a case in which the suggested delimiter order does not work properly. My apologies.

**Comment by edemaine** [comment]
This issue should now be fixed thanks to #2679; see [this test](https://github.com/KaTeX/KaTeX/blob/b34175bd92b746e22d2ca9aaf078f389efc02c17/contrib/auto-render/test/auto-render-spec.js#L148).
